### PR TITLE
Update scripts to use the Github API 

### PR DESCRIPTION
**Pull request by dibyam1101**
In this PR:
- Updated the Script to use the Github API instead of scraping data (which can potentially break when Github changes their website structure.)
- Updated the relevant tests.

### CODE REVIEW

File: new-website/utils/tutorials/fetch_tutorials.py [hunk 1]
```diff
@@ -12,83 +12,102 @@
 import os
```
Comment (rbharath) [inline new-website/utils/tutorials/fetch_tutorials.py:12]: Let's just update the README to explain that we use the github API and no longer use scraping

**Review by rbharath — COMMENTED** [review]
Almost ready to merge, small docs improvement

**Review by rbharath — APPROVED** [review]
LGTM

@dibyam1101 To fix comments in follow up PR

File: new-website/utils/tutorials/fetch_tutorials.py (hunk 1 above)
Comment (dibyam1101) [reply new-website/utils/tutorials/fetch_tutorials.py:12]: Updated the corresponding README

**Pull request merged**